lib/Sema/CSBindings.cpp

```diff
@@ -1484,6 +1484,23 @@ PotentialBindings::inferFromRelational(Constraint *constraint) {
         AdjacentVars.insert({typeVar, constraint});
     }
 
+    // Infer a binding from `inout $T <convertible to> Unsafe*Pointer<...>?`.
+    if (first->is<InOutType>() &&
+        first->getInOutObjectType()->isEqual(TypeVar)) {
+      if (auto pointerTy = second->lookThroughAllOptionalTypes()
+                               ->getAnyPointerElementType()) {
+        if (!pointerTy->isTypeVariableOrMember()) {
+          // The binding is as a fallback in this case because $T could
+          // also be Array<X> or C-style pointer.
+          if (constraint->getKind() >= ConstraintKind::ArgumentConversion)
+            DelayedBy.push_back(constraint);
+
+          return PotentialBinding(pointerTy, AllowedBindingKind::Subtypes,
```

Comment:
Could this use `AllowedBindingKind::Exact`?

Reply:
We can use that indeed!

Reply:
✅